### PR TITLE
Feat/playlist mgmt

### PR DESCRIPTION
**Pull request by bhavyaj12**
### Added Playlist Management

- Modified user data context+reducer for playlists
- Added playlist utility api functions
- Added Playlist Modal for add/delete ops for video to playlist
- Add new playlist from Playlist Modal
- All Playlists page with option to add and delete playlists
- Single playlist page displaying all videos in playlist with option to delete video
- User can remove videos from liked and watch later

**Please check app is RESPONSIVE and with theme change functionality**
**Ignore console logs, I'll remove them later**

### What not to check
- Single video page

playlist-pr

### CODE REVIEW

**Review by arpit01923 — APPROVED** [review]
- clean code
- UI works fine
- good to merge

File: src/components/PlaylistCard/PlaylistCard.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import { useNavigate } from "react-router-dom";
+import CancelScheduleSendIcon from "@mui/icons-material/CancelScheduleSend";
+import DeleteIcon from "@mui/icons-material/Delete";
```
Comment (Chiragmodi01) [inline src/components/PlaylistCard/PlaylistCard.jsx:3]: you can use barrel import of mui imports, there'll be a lot of them in places

File: src/components/PlaylistCard/playlist-card.css [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+.playlist-container .playlist-img {
+    height: 15rem;
+    position: relative;
+}
+
+.playlist-container.playlist-empty .playlist-img {
+    display: flex;
+    align-items: center;
+    justify-content: center;
```
Comment (Chiragmodi01) [inline src/components/PlaylistCard/playlist-card.css:31]: a util class you can make for these three properties, must be using at a lot of places

File: src/components/PlaylistModal/playlist-modal.css [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  top: 1rem;
+  right: 1rem;
+}
+
+.playlist-modal-demo-wrapper {
+  display: flex;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  right: 0;
```
Comment (Chiragmodi01) [inline src/components/PlaylistModal/playlist-modal.css:60]: i think 2 properties will be enough to bottom down position.

**Review by Chiragmodi01 — COMMENTED** [review]
- App is taking very long time to load cause of heavy images ig, better if you add a loader.
- Clicking on any category does not not redirect me anywhere, just reloads the page.
- Empty playlist card and filled one seems be of different height.
- If playlist is empty, opening single playlist page won't make sense.

- In mobile view, page is scrollable after opening hamburger menu. Either give position fixed or  place a div in background so home content won't be clickable

Overall code structure and UI looks good to me. Haven't picked minor UI fixes.
Good work

File: src/components/PlaylistCard/PlaylistCard.jsx (hunk 1 above)
Comment (bhavyaj12) [reply src/components/PlaylistCard/PlaylistCard.jsx:3]: yes will refactor this at the end

**Pull request merged**